pd/src/state.rs

```diff
@@ -496,18 +498,27 @@ ON CONFLICT (id) DO UPDATE SET data = $1
     }
 
     /// Retrieve the [`Asset`] for a given asset ID.
-    pub async fn asset_lookup(&self, asset_id: Vec<u8>) -> Result<Asset> {
+    pub async fn asset_lookup(&self, asset_id: Vec<u8>) -> Result<params::AssetInfo> {
         let mut conn = self.pool.acquire().await?;
 
         let asset = query!(
-            "SELECT denom, asset_id FROM assets WHERE asset_id = $1",
+            "SELECT denom, asset_id, total_supply FROM assets WHERE asset_id = $1",
             asset_id
         )
         .fetch_one(&mut conn)
         .await?;
-        Ok(Asset {
-            asset_denom: asset.denom,
-            asset_id: asset.asset_id,
+
+        // TODO: should we be returning proto types from our state methods, or domain types?
```

Comment:
Proto types are good when they're being fed into RPC responses since there's no point in validating data that's already in our database before we ship it to the client.  If we consumed it internally, we could just `.try_into().expect()` on the proto, and then we're only paying the validation cost when we need the refined type.